hls-graph/src/Development/IDE/Graph/Internal/Options.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Development.IDE.Graph.Internal.Options where
+
+import qualified Development.Shake as Shake
+import qualified Data.HashMap.Strict as Map
+import Development.IDE.Graph.Internal.Action
+import Development.IDE.Graph.Internal.Rules
+import Data.Dynamic
+
+data ShakeOptions = ShakeOptions {
+    shakeThreads :: Int,
+    shakeFiles :: FilePath,
+    shakeExtra :: Maybe Dynamic
+    }
```

Comment:
Users of ghcide as a library will want to customise shake options more extensively. 
At the very least, could expose `shakeAllowRedefineRules` too?

Reply:
Sure. I went through the full list, but only found two others seem seem remotely plausible to customise - shakeAllowRedefineRules and shakeTimings. So I've added those.